pkg/lib/ncl/proto/v1/message.proto

```diff
@@ -0,0 +1,21 @@
+syntax = "proto3";
+
+package proto.v1;
+
+option go_package = ".";
+
+message Message {
+  Metadata metadata = 1;
+  Payload payload = 2;
+}
+
+message Metadata {
+  map<string, string> fields = 1;
+  bytes checksum = 2; // checksum of the metadata
+}
+
+message Payload {
+  bytes data = 1;
```

Comment:
Is the ideology to support payload as bytes for time being ? And in future it is typed ? I don't see this giving us much value which protobuf brings to the table at the moment.

Reply:
I understand the whole serde structure and logic is not very clear. I have renamed and regrouped things in https://github.com/bacalhau-project/bacalhau/pull/4257/commits/59734b9c98aa7d2a921f3c5e9fef762d422e1e5b

This is a snipped of the new Readme:

> 1. **Message to bytes (for sending)**:
>    `Message` -> `MessageSerDe.Serialize()` -> `RawMessage` -> `EnvelopedRawMessageSerDe.Serialize()` -> `[]byte`
> 
> 2. **Bytes to Message (when receiving)**:
>    `[]byte` -> `EnvelopedRawMessageSerDe.Deserialize()` -> `RawMessage` -> `MessageSerDe.Deserialize()` -> `Message`

Let me know if anything is still not clear